### PR TITLE
feat(workflows): add weekly Poutine coverage and narrow baseline acknowledgments

### LINKED ISSUE #2983: security(workflows): complete Poutine baseline triage and main-branch coverage
Follow up on #2527 by completing the remaining Poutine work and incorporating the narrow false-positive handling prepared on `fix/poutine-setup-uv-false-positive`.

Poutine remains active as an advisory scanner in PR validation, but the repository still has no standing scan of the default branch. The current branch addresses one recurring false positive by acknowledging `astral-sh/setup-uv` only for `github_action_from_unverified_creator_used` and adding a regression test that locks that exception boundary. The broader baseline scan, finding triage, remediation, and default-branch coverage decision from #2527 remain outstanding.

## Branch Work

- Add an exact `.poutine.yml` acknowledgment for `pkg:githubactions/astral-sh/setup-uv` under `github_action_from_unverified_creator_used`.
- Add a Pester contract proving the acknowledgment has only the expected rule and package URL.
- Preserve all workflow action pins, scanner settings, and unrelated Poutine findings.
- Validate the branch and confirm its PR scan no longer reports setup-uv creator-verification findings while retaining unrelated signal.

## Remaining Work

- Run Poutine against the current default-branch commit and retain the complete SARIF result as baseline evidence.
- Triage every baseline finding as actionable, false positive, or accepted risk.
- Remediate actionable findings and use only narrowly scoped, justified acknowledgments for confirmed false positives.
- Reconcile setup-uv alerts 788 and 789 and their PR review threads after the branch scan proves the acknowledgment behaves as intended.
- Decide and implement standing default-branch coverage, such as a scheduled scan, or document why PR-only coverage is sufficient.
- Record a separate disposition for the related job-level permissions hardening opportunity noted in #2527.
- Monitor upstream Poutine issue 452 and remove the setup-uv acknowledgment once upstream creator recognition makes it obsolete.

## Acceptance Criteria

- [ ] The setup-uv acknowledgment and its regression contract are validated and merged.
- [ ] A full Poutine scan of the current default branch is captured and reviewed.
- [ ] Every baseline finding has a documented disposition.
- [ ] Actionable findings are remediated or linked to concrete follow-up issues.
- [ ] False-positive acknowledgments are scoped to the narrowest supported rule, path, job, or package identity.
- [ ] Alerts 788 and 789 and their associated review threads are verified as resolved or dispositioned after CI confirmation.
- [ ] The repository either runs Poutine on the default branch on a standing cadence or records an explicit decision to retain PR-only coverage.
- [ ] The job-level workflow-permissions hardening opportunity from #2527 is completed or split into a linked issue.
- [ ] Documentation reflects the final Poutine execution and triage model.

## Related

- Source issue: #2527
- Triggering dependency PR: #2961
- Upstream false-positive report: https://github.com/boostsecurityio/poutine/issues/452

### PR DESCRIPTION
**Pull request by jkim323**
# Pull Request

## Description

Add weekly default-branch Poutine coverage and acknowledge three evidence-backed false-positive patterns without weakening PR validation or hiding unrelated findings.

* Reuse `dangerous-workflow-scan.yml` from weekly security maintenance with both scanners advisory, SARIF/artifact uploads enabled, and explicit `contents: read` and `security-events: write` permissions. Preserve the blocking homegrown check in PR validation.
* Add exact acknowledgments for the `googleapis/release-please-action` package, the marketplace workflow's read-only `prepare-publisher` job, and the generated PR-review workflow's empty-default permissions finding. Preserve the existing acknowledgments and EndBug creator-verification signal.
* Report scan execution in the weekly summary without treating advisory job success as proof of zero findings or successful Poutine analysis.
* Add seven regression-test blocks covering selector scope, supporting controls, weekly wiring, and summary behavior for successful, failed, cancelled, and skipped scans.
* Document the complete baseline, finding dispositions, existing permissions enforcement, and conditions for removing acknowledgments.

## Related Issue(s)

Closes #2983. This PR addresses the remaining local baseline-triage and standing-coverage work. The setup-uv fix is intentionally separate; hosted confirmation and remote alert/thread reconciliation remain follow-ups, so this PR does not claim to close the issue.

## Type of Change

Select all that apply:

**Code & Documentation:**

* [x] Bug fix (non-breaking change fixing an issue)
* [x] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature causing existing functionality to change)
* [x] Documentation update

**Infrastructure & Configuration:**

* [x] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [x] Security configuration
* [ ] DevContainer configuration
* [ ] Dependency update

**AI Artifacts:**

* [ ] Reviewed contribution with `hve-builder` and addressed all actionable findings
* [ ] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [ ] Copilot agent (`.github/agents/*.agent.md`)
* [ ] Copilot skill (`.github/skills/*/SKILL.md`)
* [ ] Copilot hook (`.github/hooks/*/*.json`)
* [ ] Eval spec added/updated for changed AI artifacts (`evals/`)

> Note for AI Artifact Contributors:
>
> * Agents: Research, indexing/referencing other project (using standard VS Code GitHub Copilot/MCP tools), planning, and general implementation agents likely already exist. Review `.github/agents/` before creating new ones.
> * Skills: Must include both bash and PowerShell scripts. See [Skills](../docs/contributing/skills.md).
> * Model Versions: Contributions **MUST** target models listed in the model catalog (`scripts/linting/model-catalog.json`) whose provider appears in `providerAllowlist` and whose status is `ga` or `preview`. Run `npm run lint:models` to validate references.
> * See [Agents Not Accepted](../docs/contributing/custom-agents.md#agents-not-accepted) and [Model Version Requirements](../docs/contributing/ai-artifacts-common.md#model-version-requirements).

**Other:**

* [x] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Testing

PR-preparation checks against commit `dd5ed800`:

* `npm run test:ps -- -TestPath scripts/tests/extension/Workflow-PackagingContracts.Tests.ps1 -Tag Poutine`: 11 passed, 0 failed, 0 skipped. This is the Poutine subset, not the full packaging suite.
* `npm run lint:yaml`: all 70 workflows passed actionlint.
* `npm exec -- markdownlint-cli2 docs/security/dangerous-workflow-detection.md docs/architecture/workflows.md docs/security/security-model.md`: all three pages passed.
* `git diff --check origin/main...HEAD`: passed.

Additional checks passed during implementation of the same committed content:

* `npm run lint:ps -- -ChangedFilesOnly`: changed Pester file passed with zero issues.
* `npm run lint:permissions`: no workflow/job permission violations.
* `npm run lint:workflow-runner`: no runner-policy violations.
* `npm run lint:dangerous-workflow`: no findings.
* `npm run lint:dependency-pinning`: 767 dependencies checked, zero violations.
* `npm run lint:frontmatter -- -ChangedFilesOnly`: all three changed pages passed.
* `npm run lint:tables`: repository-wide table check passed.
* `npm run lint:md-links -- -Path docs/security/dangerous-workflow-detection.md -Quiet`: passed.
* `npm run lint:md-links -- -Path docs/architecture/workflows.md -Quiet`: passed.
* `npm run lint:md-links -- -Path docs/security/security-model.md -Quiet`: passed.

Full local/documentation aggregates, spell checking, the full packaging test suite, and browser lanes were not run. Link validation was scoped to the three changed pages. Hosted Poutine execution and the first scheduled default-branch result remain unverified.

### Required Checks

* [x] Documentation is updated (if applicable)
* [x] Files follow existing naming conventions
* [x] Changes are backwards compatible (if applicable)
* [x] Tests added for new functionality (if applicable)

### AI Artifact Contributions

* [ ] Used `hve-builder` review mode to review contribution
* [ ] Addressed all actionable findings from the `hve-builder` review
* [ ] Verified contribution follows common standards and type-specific requirements

No distributable AI artifacts or package membership changes are included.

### Required Local Checks

The following local-safe validation commands must pass before merging:

* [ ] Local validation aggregate: `npm run validate:local`
* [ ] Documentation validation (if docs changed): `npm run validate:docs`
* [ ] Spell checking: `npm run spell-check`
* [ ] Link validation: `npm run lint:md-links`

## Security Considerations

* [ ] This PR does not contain any sensitive or NDA information
* [ ] Any new dependencies have been reviewed for security issues
* [x] Security-related scripts follow the principle of least privilege

No dependency manifests or lockfiles change. The weekly scanner caller grants only repository-content read and SARIF write access, with no inherited secrets. The marketplace acknowledgment relies on the existing main-commit binding, digest verification, read-only preparation permissions, and separation from protected publication credentials. The permissions acknowledgment does not weaken the existing permissions validator or alter the generated workflow.

## Additional Notes

* The baseline contains 12 findings across three rules on a tree identical to `main` commit `5e21810e`; it is not evidence of a scan executed on the main ref.
* These acknowledgments target five baseline occurrences. Six setup-uv occurrences belong to the separate fix, and the EndBug finding remains visible. No remote alert resolution is claimed.
* Review the combined creator-rule contracts when integrating the separate setup-uv change, and rerun the Poutine tests on the resulting branch.
* After merge, confirm the first weekly scan produces current SARIF. For manual baseline collection, select `main`; manual dispatch otherwise scans the selected ref.
* Remote alert dismissal and review-thread reconciliation require current hosted evidence and separate authorization.

### CODE REVIEW

File: docs/security/dangerous-workflow-detection.md [hunk 1]
```diff
@@ -120,25 +137,81 @@ CI-only scanner and is not part of the offline lint pipeline.
...
+The job-level permissions opportunity from issue 2527 is already implemented by
+`Test-WorkflowPermissions.ps1`: absent workflow permissions fail, jobs under populated
+workflow grants need explicit permissions, and an empty workflow map grants nothing to
+jobs without overrides. No further validator or generated-workflow edit is needed here.
+
+Remove creator acknowledgments when a pinned scanner upgrade recognizes the verified
+publisher without them. Remove the PR-review acknowledgment when the scanner distinguishes
+empty permissions from missing permissions. Re-review or remove the marketplace exception
+if the checkout source, digest checks, permissions, or credential separation changes.
+Maintainers should verify each removal against hosted SARIF, not job success alone.
```
Comment (rezatnoMsirhC) [inline docs/security/dangerous-workflow-detection.md:213]: Could we move the point-in-time evidence (alert numbers, analysis `1822021547`, the synthetic merge SHA and tree hash, and the counts of occurrences handled here versus in the setup-uv change) into #2983 or the PR description? Those details will likely go stale once the alerts are reconciled or the setup-uv change merges. The page could then keep a short entry for each acknowledgment: rule, selector, reason, and removal condition. Keep the EndBug accepted-risk note as it is. That way the page stays accurate without needing follow-up edits every time an alert or related PR changes.

File: .github/workflows/weekly-security-maintenance.yml [hunk 2]
```diff
@@ -78,12 +92,15 @@ jobs:
...
           $codeqlResult = $env:CODEQL_RESULT
+          $workflowScanResult = $env:WORKFLOW_SCAN_RESULT
           $thresholdDays = $env:THRESHOLD_DAYS
           
           # Determine overall status
           $hasIssues = $false
           if ($unpinnedCount -ne '0') { $hasIssues = $true }
           if ($staleCount -ne '0') { $hasIssues = $true }
+          if ($codeqlResult -ne 'success') { $hasIssues = $true }
+          if ($workflowScanResult -ne 'success') { $hasIssues = $true }
```
Comment (rezatnoMsirhC) [inline .github/workflows/weekly-security-maintenance.yml:103]: Now that CodeQL and the dangerous-workflow scan set `$hasIssues`, a run where only one of those jobs fails, is skipped, or is cancelled will show "Next Steps" that cover only pinning and stale SHAs. Could we add conditional steps so the summary points to the job that caused the issue? For example, in the Next Steps block:

```powershell
$(if ($codeqlResult -ne 'success') { "5. CodeQL analysis result was $codeqlResult; review the codeql-analysis job logs`n" })
$(if ($workflowScanResult -ne 'success') { "6. Dangerous workflow scan result was $workflowScanResult; review the scanner step logs and confirm the poutine-results artifact exists`n" })
```

This keeps the summary actionable when neither dependency check is the cause.

File: .github/workflows/weekly-security-maintenance.yml [hunk 3]
```diff
@@ -104,8 +121,12 @@ jobs:
           | Component | Status |
           |-----------|--------|
           | CodeQL Analysis | $(if ($codeqlResult -eq 'success') { '✅ Complete' } else { '⚠️ See details' }) |
+          | Dangerous Workflow / Poutine | $workflowScanResult (advisory; review logs and SARIF) |
```
Comment (rezatnoMsirhC) [inline .github/workflows/weekly-security-maintenance.yml:124]: Nit: the other rows in this table start with a status marker, but this one shows only the raw job result. Could we add a marker too, so a non-success result stands out when scanning the report? The text stays the same, so the new Pester assertion still matches.

```suggestion
          | Dangerous Workflow / Poutine | $(if ($workflowScanResult -eq 'success') { '✅' } else { '⚠️' }) $workflowScanResult (advisory; review logs and SARIF) |
```
